_guides/managing-oauth-clients.md

```diff
@@ -0,0 +1,209 @@
+---
+title: Managing OAuth Clients
+layout: article
+section: Developing Components
+order: 1
+since: 20190403
+---
+This document reveals the new approach to OAuth utilization within the platform
+and explains how to [manage](#managing-oauth-client) OAuth clients for
+components in a tenant.
+
+## Changes
+
+
+OAuth client was defined for a component in its environment variables. This way
+global access to the component could not be enabled, because users in other tenants would see the
+component and its keys. Now OAuth clients are defined
+separately from the components’ environment variables, allowing the users in the
+other tenant to see all available components without security breach.
+
+To manage OAuth clients the user requires the following permissions:
+
+
+`tenants.oauth_clients.get`
+
+`tenants.oauth_clients.edit`
+
+`tenants.oauth_clients.create`
+
+`tenants.oauth_clients.delete`
+
+
+To acquire these permissions, please contact support.
+
+## Managing OAuth Client
+
+To initiate work with OAuth client, first specify `"useOAuthClient": true` in the _component.json_ file. For compatibility reasons, we should also create a link between the new implementation and environment variables if they were present.
```

Comment:
I'd rather to add example  where useOAuthClient is placed, because component.json is quite large
```javascript
{
  "title": "Salesforce",
  "description": "Customer relationship management (CRM) software & cloud computing from the leader in CRM solutions for businesses large & small.",
  "docsUrl": "https://github.com/elasticio/salesforce-component",
  "url": "http://www.salesforce.com/",
  "envVars": {
    "SALESFORCE_KEY": {
      "required": true,
      "description": "Your Salesforce OAuth client key"
    },
    "SALESFORCE_SECRET": {
      "required": true,
      "description": "Your Salesforce OAuth client secret"
    }
  },
  "credentials": {
    "fields": {
      "prodEnv": {
        "label": "Environment",
        "viewClass": "SelectView",
        "required": true,
        "model": {
          "test": "Sandbox",
          "login": "Production"
        },
        "prompt": "Select environment"
      },
      "oauth": {
        "label": "Authentication",
        "viewClass": "OAuthFieldView",
        "required": true
      }
    },
    "oauth2": {
      "client_id": "{{SALESFORCE_KEY}}",
      "client_secret": "{{SALESFORCE_SECRET}}",
      "auth_uri": "https://{{prodEnv}}.salesforce.com/services/oauth2/authorize",
      "token_uri": "https://{{prodEnv}}.salesforce.com/services/oauth2/token"
    },
    "useOAuthClient": true
  },
  "triggers": {
    "queryTrigger": {
      "title": "Query",
      "main": "./lib/triggers/query.js",
      "type": "polling",
      "description": "Will continuously run the same SOQL Query and emit results one-by-one",
      "metadata": {
        "out": {}
      },
      "fields": {
        "query": {
          "label": "SOQL Query",
          "required": true,
          "viewClass": "TextAreaView"
        }
      }
    },
```

Reply:
Done